lib/TelemetryFeed.jsm

```diff
@@ -63,10 +69,73 @@ this.TelemetryFeed = class TelemetryFeed {
     this._onEventsTelemetryPrefChange = this._onEventsTelemetryPrefChange.bind(this);
     this._prefs.observe(EVENTS_TELEMETRY_PREF, this._onEventsTelemetryPrefChange);
     this._classifySite = classifySite;
+    this._addWindowListeners = this._addWindowListeners.bind(this);
+    this.handleEvent = this.handleEvent.bind(this);
   }
 
   init() {
     Services.obs.addObserver(this.browserOpenNewtabStart, "browser-open-newtab-start");
+    // Add pin tab event listeners on future windows
+    Services.obs.addObserver(this._addWindowListeners, DOMWINDOW_OPENED_TOPIC);
+    // Listen for pin tab events on all open windows
+    for (let win of Services.wm.getEnumerator("navigator:browser")) {
+      if (win.closed || PrivateBrowsingUtils.isWindowPrivate(win)) {
+        continue;
+      }
+      this._addWindowListeners(win);
+    }
+  }
+
+  handleEvent(event) {
+    switch (event.type) {
+      case TAB_PINNED_EVENT:
+        this.countPinnedTab();
+        break;
+      case DOMWINDOW_UNLOAD_TOPIC:
+        this._removeWindowListeners(event.target);
+        break;
+    }
+  }
+
+  _removeWindowListeners(win) {
+    win.removeEventListener(DOMWINDOW_UNLOAD_TOPIC, this.handleEvent);
+    win.removeEventListener(TAB_PINNED_EVENT, this.handleEvent);
+  }
+
+  _addWindowListeners(win) {
```

Comment:
Perhaps move the private window check here? And only call `removeEventListener` if the window is private in the `_removeWindowListeners()`.

Reply:
That being said, it'll be nice to have a unit test for this.

Reply:
I ended up adding the events for all windows because of an issue I noticed while testing where `isWindowPrivate` returns a value based on the type of window you previously had focused.
STR
Have 2 browser windows open: one private, one regular.
Set a breakpoint in this function.
Use the keyboard shortcut or menu option to create a new _non-private_ window.
Result:
* when focus is set on the non-private window `isWindowPrivate` returns `false` for the new window as well
* when focus is set on the private window `isWindowPrivate` returns `true` for the new window as well

I assume the new window inherits attributes from the previous when it is initialized and those get changed later.
Even more interesting: there is also CMD+Shift+N to create a new window and it does not have this issue. But the shortcut shown in the browser menu is CMD+N.